service-api/module/Application/src/Model/Entity/CaseData.php

```diff
@@ -69,4 +69,16 @@ public function isValid(): bool
             ->setData(get_object_vars($this))
             ->isValid();
     }
+
+    public function toArray(): array
```

Comment:
At a minimum we should define the array shape for Psalm. Explicit getters (`getPersonType`) would be better still, but array shape will at least catch errors.
```suggestion
    /**
     * @returns array{personType: "donor"|"certificateProvider", firstName: string, lastName: string, dob: string, address: string[], lpas: string[]}
     */
    public function toArray(): array
```